src/reflect/scala/reflect/internal/Scopes.scala

```diff
@@ -8,12 +8,21 @@ package reflect
 package internal
 
 import scala.annotation.tailrec
+import scala.collection.generic.Clearable
 import scala.reflect.internal.util.Statistics
 
 trait Scopes extends api.Scopes { self: SymbolTable =>
 
   private[scala] var scopeCount = 0
 
+  // Reset scopeCount per every run
+  perRunCaches.recordCache(new Clearable {
```

Comment:
I tried -- it didn't work because the compiler failed to create a SAM type because `recordCache`'s signature expects `T <: Clearable`. It seems SAM creation doesn't work for these cases (or so I think).

Reply:
ah right, you'd need a type annotation (`(() => { scopeCount = 0 }): Clearable`)

Reply:
What I ended up doing locally is creating a new private val with the closure. IMO it's better than the type annotation. If you don't have any objection, I'll continue this way.

Reply:
Either way is fine. A local val would be better than a private val in `Scopes`, as the latter creates a field.